tests/net/lib/coap_client/src/main.c

```diff
@@ -659,6 +674,36 @@ ZTEST(coap_client, test_separate_response)
 	zassert_equal(last_response_code, COAP_RESPONSE_CODE_OK, "Unexpected response");
 }
 
+ZTEST(coap_client, test_separate_response_lost)
+{
+	int ret = 0;
+	struct k_sem sem;
+	struct sockaddr address = {0};
+	struct coap_client_request client_request = {
+		.method = COAP_METHOD_GET,
+		.confirmable = true,
+		.path = test_path,
+		.fmt = COAP_CONTENT_FORMAT_TEXT_PLAIN,
+		.cb = coap_callback,
+		.payload = short_payload,
+		.len = strlen(short_payload),
+		.user_data = &sem,
+	};
+
+	zassert_ok(k_sem_init(&sem, 0, 1));
+	z_impl_zsock_recvfrom_fake.custom_fake = z_impl_zsock_recvfrom_custom_fake_only_ack;
+	set_socket_events(client.fd, ZSOCK_POLLOUT);
+
+	k_sleep(K_MSEC(1));
+
+	LOG_INF("Send request");
+	ret = coap_client_req(&client, 0, &address, &client_request, NULL);
+	zassert_true(ret >= 0, "Sending request failed, %d", ret);
+
+	zassert_ok(k_sem_take(&sem, K_MSEC(COAP_SEPARATE_TIMEOUT * 100)));
```

Comment:
6000 * 100 msec timeout (10 minutes), was that intended? Seems quite lengthy for a test case.

Reply:
Oops.. left over from debug session. It should be just `COAP_SEPARATE_TIMEOUT`, will fix.